patches/comfyui_3d_pack.diff

```diff
@@ -1,3 +1,218 @@
++                        width,
++                        timestep,
++                        batch_size,
++                        num_images_per_prompt,
++                        dtype,
++                        device,
++                        generator=None):
++        batch_size = batch_size * num_images_per_prompt
++        if image is None:
++            return super().prepare_latents(
++                    batch_size,
++                    self.unet.config.out_channels,
++                    height,
++                    width,
++                    dtype,
++                    device,
++                    generator,
++                )
++
++        if not isinstance(image, (torch.Tensor, PIL.Image.Image, list)):
++            raise ValueError(
++                f"`image` has to be of type `torch.Tensor`, `PIL.Image.Image` or list but is {type(image)}"
++            )
++
++        image = image.permute(0, 3, 1, 2)
++        image = image.to(device=device, dtype=dtype)
++        if isinstance(generator, list) and len(generator) != batch_size:
++            raise ValueError(
++                f"You have passed a list of generators of length {len(generator)}, but requested an effective batch"
++                f" size of {batch_size}. Make sure the batch size matches the length of the generators."
++            )
++
++        if isinstance(generator, list):
++            init_latents = [
++                self.vae.encode(image[i:i + 1]).latent_dist.sample(
++                    generator[i]) for i in range(batch_size)
++            ]
++            init_latents = torch.cat(init_latents, dim=0)
++        else:
++            init_latents = self.vae.encode(image).latent_dist.sample(generator)
++
++        init_latents = 0.18215 * init_latents  # https://github.com/huggingface/diffusers/issues/437
++
++        if batch_size > init_latents.shape[
++                0] and batch_size % init_latents.shape[0] == 0:
++            # expand init_latents for batch_size
++            additional_image_per_prompt = batch_size // init_latents.shape[0]
++            init_latents = torch.cat([init_latents] *
++                                     additional_image_per_prompt,
++                                     dim=0)
++        elif batch_size > init_latents.shape[
++                0] and batch_size % init_latents.shape[0] != 0:
++            raise ValueError(
++                f"Cannot duplicate `image` of batch size {init_latents.shape[0]} to {batch_size} text prompts."
++            )
++        else:
++            init_latents = torch.cat([init_latents], dim=0)
++
++        shape = init_latents.shape
++        noise = randn_tensor(shape,
++                             generator=generator,
++                             device=device,
++                             dtype=dtype)
++
++        # get latents
++        init_latents = self.scheduler.add_noise(init_latents, noise, timestep)
++        latents = init_latents
++
++        return latents
++
```

Comment:
Instead of copy-pasting prepare latents, weren't that possible to do it like this (much less changes required): https://discord.com/channels/878201015424393246/1239916944275017819/1308813032054591614